twenty-first/src/util_types/merkle_tree.rs

```diff
@@ -608,8 +656,15 @@ pub mod merkle_tree_test {
         #[strategy(arb())]
         pub tree: MerkleTree,
 
-        #[strategy(vec(0..#tree.num_leafs(), 0..#tree.num_leafs()))]
-        pub selected_indices: Vec<usize>,
+        #[
+            strategy(
+                vec(
+                    (0 as MerkleTreeLeafIndex)..#tree.num_leafs(),
+                    0..(#tree.num_leafs() as usize)
+                )
+            )
+        ]
```

Comment:
There's no need for the first cast, since `#tree.num_leafs()` is already of type `MerkleTreeLeafIndex`.

```suggestion
        #[strategy(vec(0..#tree.num_leafs(), 0..(#tree.num_leafs() as usize)))]
```

Reply:
Doesn't proptest complain about ranges whose type is only determined after the type inference has run? I'm happy to give this commit a try but if it ends up being reverted you know why ;-)